crates/evm/src/lib.rs

```diff
@@ -64,6 +66,24 @@ impl PendingTransaction {
     }
 }
 
+impl Encodable for PendingTransaction {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        self.transaction.encode(out);
+        self.receipt.encode(out);
+    }
+}
+
+impl Decodable for PendingTransaction {
+    fn decode(rlp: &mut &[u8]) -> Result<Self, alloy_rlp::Error> {
+        let transaction = TransactionSignedAndRecovered::decode(rlp)?;
+        let receipt = Receipt::decode(rlp)?;
+        Ok(Self {
+            transaction,
+            receipt,
+        })
+    }
+}
```

Comment:
are we sure we can just stitch together two rlp's like this?

Reply:
why not?

Reply:
please make sure this is not a problem. read about how RLP encoding / decoding work. make sure this is not a problem.